api/src/client/ApiTaxClearanceCertificateClient.test.ts

```diff
@@ -0,0 +1,164 @@
+import { ApiTaxClearanceCertificateClient } from "@client/ApiTaxClearanceCertificateClient";
+import { TaxClearanceCertificateClient } from "@domain/types";
+import { DummyLogWriter } from "@libs/logWriter";
+import { LookupTaxClearanceCertificateAgenciesById } from "@shared/taxClearanceCertificate";
+import {
+  generateBusiness,
+  generateTaxClearanceCertificateData,
+  generateUser,
+  generateUserData,
+} from "@shared/test";
+import { UserData } from "@shared/userData";
+import axios from "axios";
+
+jest.mock("axios");
+const mockAxios = axios as jest.Mocked<typeof axios>;
+
+describe("TaxClearanceCertificateClient", () => {
+  const config = {
+    orgUrl: "www.test.com",
+    userName: "fakeUserName",
+    password: "fakePassword",
+  };
+  let client: TaxClearanceCertificateClient;
+  let userData: UserData;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    client = ApiTaxClearanceCertificateClient(DummyLogWriter, config);
+    userData = generateUserData({});
+    mockAxios.post.mockResolvedValue({ data: {} });
+  });
+
+  it("makes a post request to correct url with basic auth and data", () => {
+    const userData = generateUserData({
+      currentBusinessId: "123",
+      user: generateUser({
+        id: "user-id1",
+        name: "user-name1",
+      }),
+      businesses: {
+        123: generateBusiness({
+          taxClearanceCertificateData: generateTaxClearanceCertificateData({
+            requestingAgencyId: "newJerseyBoardOfPublicUtilities",
+            businessName: "my business name",
+            addressLine1: "123 main st",
+            addressLine2: "apt 1",
+            addressCity: "atlanta",
+            addressState: { shortCode: "AK", name: "Alaska" },
+            addressZipCode: "12345",
+            taxId: "123456789101",
+            taxPin: "1234",
+          }),
+        }),
+      },
+    });
+
+    client.postTaxClearanceCertificate(userData);
+    expect(mockAxios.post).toHaveBeenCalledWith(
+      "www.test.com//TYTR_ACE_App/ProcessCertificate/businessClearance",
+      {
+        repId: "user-id1",
+        repName: "user-name1",
+        taxpayerId: "123456789101",
+        taxpayerPin: "1234",
+        taxpayerName: "my business name",
+        addressLine1: "123 main st",
+        addressLine2: "apt 1",
+        city: "atlanta",
+        state: "AK",
+        zip: "12345",
+        agencyName: LookupTaxClearanceCertificateAgenciesById("newJerseyBoardOfPublicUtilities").name,
+      },
+      {
+        auth: { password: "fakePassword", username: "fakeUserName" },
+      }
+    );
+  });
+
+  it("should spy on GetId, return a value and invoke LogInfo", () => {
```

Comment:
It looks like this is the first place where we have a test to assert a log is firing. I'm not opposed, but it is a new pattern.

I might change the test name to something explicitly stating the logs should fire.

Reply:
I decided to create some tests around logging to get more familiar with spying. I’ve consolidated them into a single test and updated the name.